datasets/th/cabinet/crawler.py

```diff
@@ -0,0 +1,127 @@
+import re
+from lxml import html
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.zyte_api import fetch_html
+
+# from zavod.shed.trans import (
+#     apply_translit_full_name,
+#     make_position_translation_prompt,
+# )
+
+# NAME_PROMPT = """
+# Translate and transliterate the following Thai names into English. Provide a JSON object
+# with the fields described below for each name:
+# - `"original_name"`: The original Thai name
+# - `"translated_name"`: The name translated into English
+
+# Ensure that the translation reflects common English equivalents for titles and honorifics.
+# Consider any prefixes, titles, or components that may be present in the Thai names.
+
+# **Output Example:**
+
+# ```json
+# {
+#   "original_name": "นายอนุทิน ชาญวีรกูล",
+#   "transliterated_name": "Anutin Charnvirakul"
+# }
+# ```
+
+# Focus on providing both an accurate English translation and common romanized form where applicable,
+# maintaining standard conventions in English nomenclature.
+
+# """
+# TRANSLIT_OUTPUT = {"eng": ("Latin", "English")}
+# POSITION_PROMPT = prompt = make_position_translation_prompt("tha")
+ROLE_PATTERNS = re.compile(
+    r"(?P<name>.+?)\s*(?P<role>รองนายกรัฐมนตรี.*|รัฐมนตรีว่าการ.*|รัฐมนตรีประจำ.*|รัฐมนตรีช่วยว่าการ.*|นายกรัฐมนตรี)"
+)
+# Pattern specifically for "นายกรัฐมนตรี"
+PRIME_MINISTER_PATTERN = re.compile(r"(?P<role>นายกรัฐมนตรี)\s*(?P<name>.*)")
+
+
+def unblock_validator(doc: html.HtmlElement) -> bool:
+    return (
+        len(
+            doc.xpath(
+                ".//div[contains(@class, 'col-lg-2 col-md-2 col-sm-6 col-xs-12')]"
+            )
+        )
+        > 0
+    )
+
+
+def crawl(context: Context):
+    doc = fetch_html(
+        context, context.data_url, unblock_validator=unblock_validator, cache_days=3
+    )
+    main_container = doc.find(".//div[@class='wonderplugintabs-panel-inner']")
+
+    prime_minister = main_container.find(
+        ".//div[@style='min-height: 480; max-height: 600; display: flex; align-items: center; flex-direction: column; justify-content: center; width: 100%;']"  # to be updated
```

Comment:
this selector is super specific and likely to break. is there some way to select things here independent of their CSS properties?